lib/wasi/src/bin_factory/module_cache.rs

```diff
@@ -0,0 +1,284 @@
+use std::{cell::RefCell, collections::HashMap, ops::DerefMut, path::PathBuf, sync::RwLock};
+
+use bytes::Bytes;
+use wasmer::{AsEngineRef, Module};
+use wasmer_wasi_types::wasi::Snapshot0Clockid;
+
+use super::BinaryPackage;
+use crate::{syscalls::platform_clock_time_get, VirtualTaskManager, WasiRuntimeImplementation};
+
+pub const DEFAULT_COMPILED_PATH: &'static str = "~/.wasmer/compiled";
+pub const DEFAULT_WEBC_PATH: &'static str = "~/.wasmer/webc";
+pub const DEFAULT_CACHE_TIME: std::time::Duration = std::time::Duration::from_secs(30);
```

Comment:
This code should not live here. For two main reasons:
1. It couples a given path when people may be using `wasmer-wasi` as a library
2. It's tied into other logic (server/deploy) that should probably live there

Reply:
Tracked in https://github.com/wasmerio/wasmer/issues/3603